### PR TITLE
Feature/analysis module setup

### PR DESCRIPTION
**Pull request by AdityaKhatri**

### CODE REVIEW

**Review by subinasr — APPROVED** [review]

**Review by samshara — APPROVED** [review]

File: .babelrc.js [hunk 1]
```diff
@@ -45,6 +45,7 @@ module.exports = {
                 'extensions': ['.js', '.jsx', '.ts', '.tsx'],
                 'alias': {
                     '#components': './src/components',
+                    '#dui': './src/components/ui',
```
Comment (tnagorra) [inline .babelrc.js:48]: Why is this named `#dui` ?

Comment (tnagorra) [reply .babelrc.js:48]: Like "THE UI"?

File: src/typings/analysisModule.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export interface AnalysisElement {
```
Comment (tnagorra) [inline src/typings/analysisModule.tsx]: Create a common interface name Entity;
```
interface Entity = {...}
export type AnalysisElement = Entity;
export type PillarAnalysisElement = Entity;
```

File: src/constants/routes.js [hunk 3]
```diff
@@ -363,6 +374,25 @@ export const routes = {
...
+            exact: true,
+        });
+
+        if (match) {
+            return match;
+        }
+    }
+
+    return null;
+};
```
Comment (tnagorra) [inline src/constants/routes.js]: OK but why?

File: src/constants/routes.js [hunk 4]
```diff
@@ -363,6 +374,25 @@ export const routes = {
 export const pathNames = mapObjectToObject(routes, route => route.path);
 export const validLinks = mapObjectToObject(routes, route => route.links);
 export const hideNavbar = mapObjectToObject(routes, route => !!route.hideNavbar);
+export const showSubNavbar = mapObjectToObject(routes, route => !!route.showSubNavbar);
+
+export const getCurrentMatch = (location) => {
+    const links = Object.keys(pathNames);
+    const paths = Object.values(pathNames);
```
Comment (tnagorra) [inline src/constants/routes.js]: You can iterate properly using Object.entries

File: src/Multiplexer/index.js [hunk 5]
```diff
@@ -134,27 +141,48 @@ export default class Multiplexer extends React.PureComponent {
...
-        );
-    }
+    }, [authenticated]);
+
+    const [parentNode, setParentNode] = useState(null);
+    const currentMatch = useMemo(() => getCurrentMatch(location), [location]);
+
+    const currentPath = useMemo(() => (
+        isDefined(currentMatch)
+            ? getFirstKeyByValue(pathNames, currentMatch.path) : 'fourHundredFour'
```
Comment (tnagorra) [inline src/Multiplexer/index.js:151]: Why sir why?

File: src/components/general/Navbar/index.js [hunk 6]
```diff
@@ -225,30 +197,36 @@ export default class Navbar extends React.PureComponent {
...
+        ) !== -1;
+        return validProjectId;
+    }, [activeProject, userProjects]);
+
+    const [,, triggerChangeActiveProject] = useRequest({
+        url: `server://users/${userId}/`,
+        method: 'PATCH',
+        body: ({
+            lastActiveProject: validActiveProjectId,
+        }),
```
Comment (tnagorra) [inline src/components/general/Navbar/index.js]: This looks dangerous. Will be called on ever re-render

File: src/components/general/Navbar/index.js [hunk 7]
```diff
@@ -354,19 +326,52 @@ export default class Navbar extends React.PureComponent {
...
                     </p>
                 </Confirm>
-            </nav>
-        );
-    }
+            </div>
+            {showSubNavbar[currentPath] && (
+                <div className={styles.subNavbar}>
+                    <div className={styles.projectSelectInputWrapper}>
+                        <Cloak
```
Comment (tnagorra) [inline src/components/general/Navbar/index.js:343]: Does showing a sub navbar add this project jump by default?

File: src/views/AnalysisModule/Analysis/index.tsx [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+                    {startDate && endDate && (
+                        <div className={styles.separator}>-</div>
+                    )}
+                    <FormattedDate
+                        title={_ts('analysis', 'endDateLabel')}
+                        value={endDate}
+                        mode="MMM yyyy"
+                        emptyComponent={emptyComponent}
+                    />
+                </>
```
Comment (tnagorra) [inline src/views/AnalysisModule/Analysis/index.tsx]: Please create a re-usable component for this.

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: .babelrc.js (hunk 1 above)
Comment (AdityaKhatri) [reply .babelrc.js:48]: deep-ui

File: src/typings/analysisModule.tsx (hunk 2 above)
Comment (AdityaKhatri) [reply src/typings/analysisModule.tsx]: This is temporary and new things will be added

File: src/views/AnalysisModule/Analysis/index.tsx (hunk 8 above)
Comment (AdityaKhatri) [reply src/views/AnalysisModule/Analysis/index.tsx]: Date Range? Makes sense. Will do.

File: src/Multiplexer/index.js (hunk 5 above)
Comment (AdityaKhatri) [reply src/Multiplexer/index.js:151]: Old logic, but does make sense right?

File: src/components/general/Navbar/index.js (hunk 7 above)
Comment (AdityaKhatri) [reply src/components/general/Navbar/index.js:343]: Yes

File: src/views/AnalysisModule/Analysis/index.tsx (hunk 8 above)
Comment (AdityaKhatri) [reply src/views/AnalysisModule/Analysis/index.tsx]: done

File: src/constants/routes.js (hunk 3 above)
Comment (AdityaKhatri) [reply src/constants/routes.js]: Was done previously, just shifted the logic to this place

File: src/constants/routes.js (hunk 4 above)
Comment (AdityaKhatri) [reply src/constants/routes.js]: old code did not change

**Pull request merged**